yew/src/agent/worker/queue.rs

```diff
@@ -0,0 +1,52 @@
+use std::any::TypeId;
+use std::cell::{RefCell, RefMut};
+use std::collections::{hash_map, HashMap, HashSet};
+
+/// Thread-local instance used to queue worker messages
+pub struct Queue {
+    loaded_agents: RefCell<HashSet<TypeId>>,
+    msg_queue: RefCell<HashMap<TypeId, Vec<Vec<u8>>>>,
+}
+
+impl Queue {
+    pub fn new() -> Queue {
+        Queue {
+            loaded_agents: RefCell::new(HashSet::new()),
+            msg_queue: RefCell::new(HashMap::new()),
+        }
+    }
+
+    #[inline]
+    pub fn borrow_msg_queue_mut(&self) -> RefMut<'_, HashMap<TypeId, Vec<Vec<u8>>>> {
+        self.msg_queue.borrow_mut()
+    }
+
+    #[inline]
+    pub fn insert_loaded_agent(&self, type_id: TypeId) {
+        self.loaded_agents.borrow_mut().insert(type_id);
+    }
+
+    #[inline]
+    pub fn is_worker_loaded(&self, type_id: &TypeId) -> bool {
+        self.loaded_agents.borrow().contains(type_id)
+    }
+
+    pub fn add_msg_to_queue(&self, msg: Vec<u8>, type_id: TypeId) {
```

Comment:
One issue with using `TypeId` of the worker is that there may be many `Private` workers spawned, as each time `spawn_or_join` is called, a new instance is spawned. With this implementation, multiple instances will share a single message queue, leading to funky unintended behaviors.

Reply:
I've been trying to think about a good solution for this instead of introducing a unique identifier for the private agents.

If all public agents share the same queue, I think the best direction might be to turn the `loaded_agent` into a HashSet of a new Struct that'd include both the `type_id` as well as another unique `identifier` to keep private agents seperated, while the public agents share their identifier. -- Do you think that could work?

Reply:
I think generics would probably be best.. so public agents would use `TypeId` and private agents would use a unique identifier. Then the queue implementation stays simple, thoughts?